src/charm.py

```diff
@@ -714,19 +750,36 @@ def _is_bessd_grpc_service_ready(self) -> bool:
     def _is_bessd_configured(self) -> bool:
         """Check if bessd has been configured.
 
-        Examine the output from bessctl to show worker. If there is no
-        active worker, bessd is assumed not to be configured.
+        Bess is considered as configured when the `accessRoutes` and `coreRoutes` modules
+        are available and at least one worker is in the RUNNING state. This method examines
+        the output of the `bessctl show` command to determine whether the above criteria are met.
 
         Returns:
-            bool:   True/False
+            bool: Whether the `bessd` service is configured
         """
-        command = "/opt/bess/bessctl/bessctl show worker"
+        show_accessRoutes_module_cmd = "/opt/bess/bessctl/bessctl show module accessRoutes"  # noqa: N806
+        show_coreRoutes_module_cmd = "/opt/bess/bessctl/bessctl show module coreRoutes"  # noqa: N806
+        show_worker_cmd = "/opt/bess/bessctl/bessctl show worker"
         try:
```

Comment:
There is quite a lot inside the same try/except, should we separate those in smaller try/excepts to better catch exceptions?

Reply:
As discussed, since all these commands throw the same exception I think it makes sense to keep them under one `try-except` block. We also log the exact error message, so we'll know which check failed.
One thing I've changed is the log level for the error message. It's now `error`, so it'll be clearly visible in the debug-log.